### DISCUSSION TITLE
Pass a list of object as a prop in remark-directive

### DISCUSSION THREAD

**Discussion by edward1127** [remarkjs/remark#1185] [Q&A] [answered]
Is it possible to pass a list of object as as prop in remark-directive? 

e.g. 

`::list-element{items="[{'label': 'apple', 'value': 'apple'},{'label': 'banana', 'value': 'banana'}]"}`

**Comment by wooorm** [comment, marked as answer]
You can do that the same way you can do that in HTML: pass JSON in at attribute. Parse the JSON on the other side.

**Reply by edward1127** [reply]
Can it be parsed automatically? If not possible, I will add a preprocessing step as you suggested. thanks for replying

**Reply by wooorm** [reply]
Define “automatically”? Because nothing here is done by default, everything here is “automatically” done by plugins. Directives don’t do anything without other plugins. And custom elements in HTML don’t do anything by default. So in your other plugin, handle JSON, if you want it!

**Reply by edward1127** [reply]
got it. thanks!
